python-sdk/pachyderm_sdk/api/cdr/resolver.py

```diff
@@ -0,0 +1,147 @@
+"""Handwritten classes/methods that augment the existing CDR API."""
+
+import gzip
+from hashlib import blake2b
+from hmac import compare_digest
+from urllib.parse import urlparse
+
+from betterproto import which_one_of
+
+from . import (
+    Ref,
+    Cipher,
+    Compress,
+    Concat,
+    ContentHash,
+    Http,
+    SizeLimits,
+    Slice,
+    CipherAlgo,
+    CompressAlgo,
+    HashAlgo,
+)
+
+try:
+    import requests
+    from Crypto.Cipher import ChaCha20
```

Comment:
Two things:
1. Better error message for the user.
2. This will (probably) be necessary for the second half of this implementation where the `storage` API gets extended. That code will need to import the code in this file and when the API gets instantiated on the Client it would raise import errors for users who don't need CDR support.

Reply:
ah gotcha, that makes sense